### PR TITLE
Release notes for 1.67.0

### PR DESCRIPTION
**Pull request by cuviper**
r? @Mark-Simulacrum 
cc @rust-lang/release

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -1,3 +1,101 @@
+Version 1.67.0 (2023-01-26)
+==========================
+
+<a id="1.67.0-Language"></a>
```
Comment (cuviper) [inline RELEASES.md:4]: These new anchors (https://github.com/rust-lang/relnotes/pull/111) do work, but they don't change the link you get when you hover over the section header. Not sure if that's expected -- cc @pnkfelix.

Comment (ehuss) [reply RELEASES.md:4]: AFAIK, changing the link isn't possible with GitHub flavored markdown.

File: RELEASES.md [hunk 2]
```diff
@@ -1,3 +1,101 @@
...
+Refer to Rust's [platform support page][platform-support-doc]
+for more information on Rust's tiered platform support.
+
+<a id="1.67.0-Libraries"></a>
+
+Libraries
+---------
+
+- [Merge `crossbeam-channel` into `std::sync::mpsc`.](https://github.com/rust-lang/rust/pull/93563/)
+- [Fix inconsistent rounding of 0.5 when formatted to 0 decimal places.](https://github.com/rust-lang/rust/pull/102935/)
```
Comment (joshtriplett) [inline RELEASES.md:42]: This should be listed under compatibility notes, even though it's minor and unlikely to affect people.

**Comment by cuviper** [comment]
@joshtriplett I also moved the layout change to compat notes, as you had requested on Zulip. Please let me know what you think of that description.

**Pull request merged**